common/json.hpp

```diff
@@ -18,23 +18,85 @@
 #include <common/error.hpp>
 #include <common/expected.hpp>
 
-namespace json {
+#include "config.h"
+
+#ifdef MENDER_COMMON_USE_NLOHMANN_JSON
+#include <nlohmann/json.hpp>
+#endif
+
+using namespace std;
+
+namespace mender::common::json {
 
 enum JsonErrorCode {
 	NoError = 0,
-	KeyNoExist,
+	ParseError,
+	KeyError,
+	IndexError,
 	TypeError,
 };
 using JsonError = mender::common::error::Error<JsonErrorCode>;
 
 class Json {
 public:
 	using ExpectedJson = mender::common::expected::Expected<Json, JsonError>;
-	void hello_world();
+	using ExpectedString = mender::common::expected::Expected<string, JsonError>;
+	using ExpectedInt = mender::common::expected::Expected<int, JsonError>;
+	using ExpectedBool = mender::common::expected::Expected<bool, JsonError>;
+	using ExpectedSize = mender::common::expected::Expected<size_t, JsonError>;
```

Comment:
Are these needed when you already have the ones at the bottom of the file?

Reply:
It didn't compile when I tried because those are below this class definition. And for `ExpectedJson` it's a practically a circular dependency.

Reply:
I move the ones that don't refer to the `Json` class before the `class Json` definition so that at least those don't have to be duplicated.